gnome-initial-setup/pages/keyboard/cc-input-chooser.c

```diff
@@ -178,12 +189,40 @@ preview_cb (GtkLabel       *label,
 		return TRUE;
 
 	if (variant[0])
-		commandline = g_strdup_printf ("gkbd-keyboard-display -l \"%s\t%s\"", layout, variant);
+		commandline = g_strdup_printf (GKBD_KEYBOARD_DISPLAY_CMD" -l \"%s\t%s\"", layout, variant);
 	else
-		commandline = g_strdup_printf ("gkbd-keyboard-display -l %s", layout);
-	g_spawn_command_line_async (commandline, NULL);
-	g_free (commandline);
+		commandline = g_strdup_printf (GKBD_KEYBOARD_DISPLAY_CMD" -l %s", layout);
+
+	g_shell_parse_argv(commandline,
+			   &argc,
+			   &argv,
+			   &error);
+
+	if (error) {
+		g_critical ("Error parsing gkbd-keyboard-display command: %s", error->message);
+		g_clear_error (&error);
+		goto out;
+	}
 
+	g_spawn_async (NULL,
+		       argv,
+		       env,
+		       G_SPAWN_DEFAULT,
+		       NULL,
+		       NULL,
+		       &pid,
+		       &error);
+
+	if (error) {
+		g_critical ("Error running gkbd-keyboard-display: %s", error->message);
+		g_clear_error (&error);
+		goto out;
+	}
+
+out:
+	g_strfreev (env);
+	g_strfreev (argv);
```

Comment:
`argv` is possibly uninitialized here, if `g_shell_parse_argv()` failed. Since `g_strfreev()` is `NULL`-safe, you can just initialize it to make it safe.